hollow/src/main/java/com/netflix/hollow/api/producer/ListenerSupport.java

```diff
@@ -169,4 +182,16 @@ void fireAnnouncementComplete(ProducerStatus.Builder psb) {
         ProducerStatus st = psb.build();
         for(final HollowProducerListener l : listeners) l.onAnnouncementComplete(st, psb.elapsed(), MILLISECONDS);
     }
+
+    void fireIncrementalCycleComplete(long version, long recordsAddedOrModified, long recordsRemoved, HashMap<String, Object> cycleMetadata) {
+        IncrementalCycleStatus.Builder icsb = new IncrementalCycleStatus.Builder().success(version, recordsAddedOrModified, recordsRemoved);
```

Comment:
This is where an `icsb.withMetadata(cycleMetadata)` would go instead of passing it directly to listener callback.